src/core/SpyRegistry.js

```diff
@@ -175,18 +175,34 @@ getJasmineRequireObj().SpyRegistry = function(j$) {
         prop,
         descriptor;
 
-      while (pointer) {
-        for (prop in pointer) {
-          if (
-            Object.prototype.hasOwnProperty.call(pointer, prop) &&
-            pointer[prop] instanceof Function
-          ) {
-            descriptor = Object.getOwnPropertyDescriptor(pointer, prop);
-            if (
-              (descriptor.writable || descriptor.set) &&
-              descriptor.configurable
-            ) {
-              props.push(prop);
+      while (
+        pointer &&
+        (!includeNonEnumerable || pointer !== Object.prototype)
+      ) {
+        var enumerableProperties = Object.keys(pointer);
+        var properties = includeNonEnumerable
+          ? Object.getOwnPropertyNames(pointer).filter(function(prop) {
+              return (
+                prop !== 'constructor' ||
+                enumerableProperties.includes('contructor')
+              );
+            })
+          : enumerableProperties;
+        for (var i = 0; i < properties.length; i++) {
+          prop = properties[i];
+          if (Object.prototype.hasOwnProperty.call(pointer, prop)) {
+            try {
```

Comment:
Big try-catch-ignore blocks are scary because it's easy to end up ignoring more exceptions than intended. I'd rather see the try block scoped to the single expression that's expected to throw the exception. It would also be good to have a test for the scenario that the try-catch supports.

Reply:
Try-catch is over only the line where we access the property now.